src/js/controllers/RegistrationController.js

```diff
@@ -0,0 +1,52 @@
+function RegistrationController($scope, $http, $cookies, $state) {
+    $scope.registerSubmitting = false;
+
+    $scope.register = () => {
+        $scope.registerForm = {general: {}, email: {}, password: {}, confirm: {}};
+        $scope.registerSubmitting = true;
+
+        // Do some simple input validation.
+        if (!$scope.account) {
+            console.log('no input.');
+            $scope.registerForm.general['no-input'] = true;
+            $scope.registerSubmitting = false;
+            return;
+        } else if (!$scope.account.password || $scope.account.password.length === 0) {
+            $scope.registerForm.password.required = true;
+            $scope.registerSubmitting = false;
+            return;
+        } else if ($scope.account.password !== $scope.account.confirm) {
+            $scope.registerForm.confirm.mismatch = true;
+            $scope.registerSubmitting = false;
+            return;
+        }
+
+        try {
+            $http.post('http://api.kitchen.support/accounts/create',
+                {email: $scope.account.email, password: $scope.account.password}
```

Comment:
My preferred code style for this would be something like:
```javascript
$http.post('url', {
    email: blah,
    password: blah
}).success(() => {});
```